Add environment_ids attribute to project resource

bengesoff opened the pull request:
This PR adds the `environment_ids` attribute to both the project resource and data source. This depends on the associated SDK PR https://github.com/skytap/skytap-sdk-go/pull/61. With this attribute, the project resource can be used to add/remove environments from the project from within terraform.

The equivalent `project_ids` field can't be added to the environment resource because, due to the order in which Terraform updates things, this won't be updated in the same `terraform apply` as the project resource which is confusing to the user. Therefore it was decided that the `environment_ids` on the project resource was a more natural way of representing things and that this should be implemented over the other.

bengesoff committed:
Update skytap-sdk-go to include new AddEnvironment API calls

bengesoff committed:
Add environment_ids attribute to project resource and data source

bengesoff committed:
Update documentation for environment_ids fields

trentrosenbaum reviewed (approved):
Approved waiting for the Skytap Go SDK PR to be applied.

bengesoff committed:
Update skytap-sdk-go to latest, removing temporary reference to fork

The go.mod had a temporary replace directive to get the skytap-sdk-go dependency from the opencredo fork before the PR was merged.
The PR has now been merged so the go.mod can reference the main upstream repo again.

bengesoff commented:
The associated SDK PR has been merged so I've updated the reference, rebased onto master, and force pushed it back up
